pandas/_libs/parsers.pyi

```diff
@@ -0,0 +1,77 @@
+from typing import (
+    Any,
+    Literal,
+)
+
+import numpy as np
+
+from pandas._typing import (
+    ArrayLike,
+    Dtype,
+)
+
+STR_NA_VALUES: set[str]
+
+
+def sanitize_objects(
+    values: np.ndarray,  # ndarray[object]
+    na_values: set,
+    convert_empty: bool = ...,
+) -> int: ...
+
+
+class TextReader:
+    unnamed_cols: set[str]
+    table_width: int   # int64_t
+    leading_cols: int  # int64_t
+    header: list[list[int]]  # non-negative integers
+
+    def __init__(
+        self,
+        source,
+        delimiter: bytes | str = ...,  # single-character only
+        header=0,
+        header_start=0,
+        header_end=0,
+        index_col=...,
+        names=...,
+        tokenize_chunksize: int = ...,  # int64_t
+        delim_whitespace: bool = ...,
+        converters=...,
+        skipinitialspace: bool = ...,
+        escapechar: bytes | str | None = ...,  # single-character only
+        doublequote: bool = ...,
+        quotechar: str | bytes | None = ...,  # at most 1 character
+        quoting: int = ...,
+        lineterminator: bytes | str | None = ...,  # at most 1 character
+        comment=...,
+        decimal: bytes | str = ...,  # single-character only
+        thousands: bytes | str | None = ...,  # single-character only
+        dtype: Dtype | dict[Any, Dtype] = ...,
```

Comment:
we want to avoid using Any, maybe could use Hashable. (which is effectively Any for a dictionary key)

this fails with `error: Dict entry 0 has incompatible type "List[<nothing>]": "int"; expected "Hashable": "int"  [dict-item]`

```
from __future__ import annotations
from typing import Hashable


def func(a: dict[Hashable, int]):
    pass


func({list(): 3})
```

but this passes mypy checks

```
from __future__ import annotations
from typing import Any


def func(a: dict[Any, int]):
    pass


func({list(): 3})
```

although granted you can't create a dict with an unhashable key

```
Traceback (most recent call last):
  File "/home/simon/t.py", line 9, in <module>
    func({list(): 3})
TypeError: unhashable type: 'list'
```